### PR TITLE
making bind_name optional when bind_type is management

### LINKED ISSUE #324: nomad_acl_binding_rule requires bind_name when it should not
### Nomad Version
1.5.3

### Affected Resource(s)
Please list the resources as a list, for example:
- nomad_acl_binding_rule

### Terraform Configuration Files
```hcl
resource "nomad_acl_binding_rule" "main" {
  description = "example management rule"
  auth_method = nomad_acl_auth_method.main.name
  selector    = "admin in list.roles"
  bind_type   = "management"
}
```

### Expected Behavior
This should create an ACL binding rule that binds to a management token.

### Actual Behavior
It fails due to requiring the bind_name argument.

### Important Factoids
The Nomad documentation specifies the bind-name argument as optional, and specifically says it should not be provided when the bind type is management.

> [-bind-name](https://developer.hashicorp.com/nomad/docs/commands/acl/binding-rule/create#bind-name): Specifies is the target of the binding used on selector match. This can be lightly templated using HIL ${foo} syntax. If the bind type is set to management, this should not be set.

https://developer.hashicorp.com/nomad/docs/commands/acl/binding-rule/create

Of note, providing an empty string (`""`) does work to get around this.

### PR DESCRIPTION
**Pull request by regner**
When creating a ACL binding rule, the bind name must not be defined but only when the bind_type is set to management.

https://developer.hashicorp.com/nomad/docs/commands/acl/binding-rule/create

Fixes #324

### CODE REVIEW

**Review by pkazmierczak — APPROVED** [review]
Hey @regner, thanks for the PR!
LGTM!

File: nomad/resource_acl_binding_rule.go [hunk 1]
```diff
@@ -172,3 +182,18 @@ func generateNomadACLBindingRule(d *schema.ResourceData) *api.ACLBindingRule {
...
+	bindName := d.Get("bind_name").(string)
+	bindType := d.Get("bind_type").(string)
+
+	if bindType != api.ACLBindingRuleBindTypeManagement && bindName == "" {
+		return fmt.Errorf("error bind_name must be defined if bind_type is not '%q'", api.ACLBindingRuleBindTypeManagement)
+	}
+
+	if bindType == api.ACLBindingRuleBindTypeManagement && bindName != "" {
+		return fmt.Errorf("error bind_name must not be defined if bind_type is '%q'", api.ACLBindingRuleBindTypeManagement)
+	}
```
Comment (lgfa29) [inline nomad/resource_acl_binding_rule.go:196]: I think this is a bit easier to read?

```go
switch bindType {
case api.ACLBindingRuleBindTypeManagement:
	if bindName != "" {
		return fmt.Errorf("error bind_name must not be defined if bind_type is '%q'", api.ACLBindingRuleBindTypeManagement)
	}
default:
	if bindName == "" {
		return fmt.Errorf("error bind_name must be defined if bind_type is '%q'", bindType)
	}
}
```

**Review by lgfa29 — APPROVED** [review]
Minor nit-picking in the validation check, but not a blocker. Thanks for fixing this! 🙂

**Comment by regner** [comment]
Adjusted the validateNomadACLBindingRule formatting based on @lgfa29 feedback. I am definitely not used to reaching for switch statements so they don't come to my mind first, and I personally find it a bit harder to read, but I think that's just because I so rarely use them.

Thank you for the suggestion. I can definitely see how it would be easier to read once I train my brain a bit more.

**Pull request merged**